rita_client/src/operator_update/tests.rs

```diff
@@ -0,0 +1,260 @@
+        // from the last seen hour. we removed the last seen from usage relay, so we should expect to see our earliest hour as one fewer
+        assert!(res_usage_client.get(0).unwrap().index == last_seen_hour);
+        assert!(res_usage_relay.get(0).unwrap().index == last_seen_hour);
+
+        // now check that same thing, but in case we have a gap in the data. we'll remove the entry for the last_seen_hour from usage_data_relay
+        // to make sure we are successfully returning the next earliest hour (in our case, the last seen -1). we use res_usage client and relay
+        // because to remove the correct entry it needs to be presorted (if we've gotten here it's guaranteed.)
+        // we successfully search for the entry or return the next one down.
+        unshuffled_client.remove(unshuffled_client.len() - 11);
+        unshuffled_relay.remove(unshuffled_relay.len() - 11);
+        // so the index of our last seen hour if we say last seen is current - 10... will be at len - 11.
+        let res_usage = process_usage_data(
+            unshuffled_client,
+            unshuffled_relay,
+            last_seen_hour,
+            current_hour,
+        )
+        .unwrap();
+        let res_usage_client = res_usage.client_bandwidth;
+        let res_usage_relay = res_usage.relay_bandwidth;
+        // after processing we should start at last seen - 1.
+        println!(
+            "{:?} last seen {:?}",
+            res_usage_relay.get(0).unwrap().index,
+            last_seen_hour
+        );
+        assert!(res_usage_relay.get(0).unwrap().index == last_seen_hour - 1);
+        assert!(res_usage_client.get(0).unwrap().index == last_seen_hour - 1);
+
+        // check that our iteration function does indeed stop at a month of data:
+        assert!(res_usage_client.len() <= 730);
+        assert!(res_usage_relay.len() <= 730);
+    }
+
+    // generates a usage tracker struct for testing without payments since these do not get sent up in ops updates.
+    // using this while I can't get the import working... as a note the original function needs to be updated to push to back
+    // instead of front, as this generates data in the wrong order
+    fn generate_dummy_usage_tracker_temp() -> RCUsageTracker {
+        let current_hour = get_current_hour().unwrap();
+        RCUsageTracker {
+            last_save_hour: current_hour,
+            client_bandwidth: generate_bandwidth(current_hour),
+            relay_bandwidth: generate_bandwidth(current_hour),
+            exit_bandwidth: VecDeque::new(),
+            payments: VecDeque::new(),
+        }
+    }
+    #[cfg(test)]
+    // generates dummy usage hour data randomly
+    fn generate_bandwidth(starting_hour: u64) -> VecDeque<RCUsageHour> {
+        use rand::{thread_rng, Rng};
+        // 8760 is the max number of saved usage entries(1 year)
+        let num_to_generate: u16 = thread_rng().gen_range(50..8760);
```

Comment:
instead of using a hardcoded 8760 the constant MAX_USAGE_ENTRIES should be made public and used.